pkg/channel/fanout/fanout_message_handler.go

```diff
@@ -211,3 +281,22 @@ func (f *MessageHandler) makeFanoutRequest(ctx context.Context, message binding.
 		sub.RetryConfig,
 	)
 }
+
+type dispatchResult struct {
```

Comment:
do you really need this one? can you just use a void pointer (`interface {}`) and then do the switch type on the receiver? this should reduce of one pointer the allocations

Reply:
I can try your suggestion.

Reply:
Then validate if there is some difference looking at the allocations with the microbenchmarks :smile:

Reply:
Yeah sure I have thought about it its important not to add any overhead :)

Reply:
If there is no difference in terms of mem usage, then keep it as is

Reply:
Basically @slinkydeveloper I need this for the channel reported value, not sure I follow what you suggest. I need to use both the dispatch error and the fanout error.

Reply:
ah it's not an union type? aka or the dispatch error or the fanout error?

Reply:
Nope its weird logic there.

Reply:
Let me check again if I can simplify it.

Reply:
I removed `error` returned value and now I am using the struct for both the channel value type and for communicating the result to receiver.